File: 3-HARD/Python/5-line-sweep-closest-points/proposed-solution/main.py
# Challenge: Use the line sweep algorithm to find the closest pair of points in a set of 2D coordinates.
# This challenge requires a good understanding of geometric algorithms and spatial optimization.
# Use object-oriented programming and follow the DRY principle.

from closest_points import ClosestPoints

def main():
    points = [(2, 3), (12, 30), (40, 50), (5, 1), (12, 10), (3, 4)]
    cp = ClosestPoints(points)
    p1, p2, dist = cp.find_closest_pair()
    print(f"Closest points: {p1} and {p2} with distance {dist}")

if __name__ == "__main__":
    main()

File: 3-HARD/JavaScript/5-line-sweep-closest-points/proposed-solution/main.js
/*
Challenge:
Use the line sweep algorithm to find the closest pair of points in a set of 2D coordinates.

This solution follows DRY principles and is implemented in JavaScript.
*/

import ClosestPoints from "./ClosestPoints.js";

function main() {
  const points = [
    [2, 3],
    [12, 30],
    [40, 50],
    [5, 1],
    [12, 10],
    [3, 4],
  ];

  const closestPoints = new ClosestPoints(points);
  const result = closestPoints.findClosestPair();

  console.log("Closest pair of points:", result.pair);
  console.log("Distance between closest points:", result.distance);
}

main();
